src/renderers/dom/shared/eventPlugins/ChangeEventPlugin.js

```diff
@@ -260,34 +119,32 @@ function handleControlledInputBlur(inst, node) {
 var ChangeEventPlugin = {
   eventTypes: eventTypes,
 
-  _isInputEventSupported: isInputEventSupported,
-
-  extractEvents: function(
-    topLevelType,
-    targetInst,
-    nativeEvent,
-    nativeEventTarget,
-  ) {
+  extractEvents(topLevelType, targetInst, nativeEvent, nativeEventTarget) {
     var targetNode = targetInst
       ? ReactDOMComponentTree.getNodeFromInstance(targetInst)
       : window;
 
-    var getTargetInstFunc, handleEventFunc;
-    if (shouldUseChangeEvent(targetNode)) {
-      getTargetInstFunc = getTargetInstForChangeEvent;
-    } else if (isTextInputElement(targetNode)) {
-      if (isInputEventSupported) {
-        getTargetInstFunc = getTargetInstForInputOrChangeEvent;
-      } else {
-        getTargetInstFunc = getTargetInstForInputEventPolyfill;
-        handleEventFunc = handleEventsForInputEventPolyfill;
+
+    // On the selectionchange event, the target is the document which isn't
+    // helpful becasue we need the input, so we use the activeElement instead.
+    if (!isTextInputEventSupported && topLevelType === 'topSelectionChange') {
```

Comment:
We were already tracking the active element for this code so the constraint isn't new, i'm just deferring to `activeElement` instead of caching it locally in a variable.